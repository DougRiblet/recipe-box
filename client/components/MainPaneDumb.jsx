import React from 'react'
import marked from 'marked'
import MainPaneEdit from './MainPaneEdit'

function createMarkup (input) {
  let parsedInput = marked(input, {sanitize: true})
  return {__html: parsedInput}
}

function parseMarkupDangerously (input) {
  return <div dangerouslySetInnerHTML={createMarkup(input)} />
}

function generateRecipeDisplay (recipe, setMode) {
  return (
    <div id='contentDisplay'>
      <h1>{recipe.title}</h1>
      <div id='ing-column'>
        <h4>Ingredients</h4>
        {parseMarkupDangerously(recipe.ingredients)}
      </div>
      <div id='dir-column'>
        <h4>Directions</h4>
        {parseMarkupDangerously(recipe.directions)}
        <h4>Notes</h4>
        {parseMarkupDangerously(recipe.notes)}
      </div>
      <div className='mainPaneButtons'>
        <button
          className='setModeButton'
          onClick={() => setMode('edit')}
        >Edit Recipe</button>
      </div>
    </div>
  )
}

function generateRecipeEdit (recipe, updateRecipe, deleteRecipe, deleteAllowed, setMode) {
  return (
    <MainPaneEdit
      recipe={recipe}
      updateRecipe={updateRecipe}
      deleteAllowed={deleteAllowed}
      deleteRecipe={deleteRecipe}
      setMode={setMode}
    />
  )
}

const MainPaneDumb = ({ mode, recipe, setMode, updateRecipe, deleteRecipe, deleteAllowed }) => (
  mode === 'edit'
  ? generateRecipeEdit(recipe, updateRecipe, deleteRecipe, deleteAllowed, setMode)
  : generateRecipeDisplay(recipe, setMode)
)

MainPaneDumb.propTypes = {
  mode: React.PropTypes.string.isRequired,
  recipe: React.PropTypes.shape({
    id: React.PropTypes.string.isRequired,
    title: React.PropTypes.string,
    ingredients: React.PropTypes.string,
    directions: React.PropTypes.string,
    notes: React.PropTypes.string
  }).isRequired,
  setMode: React.PropTypes.func.isRequired,
  updateRecipe: React.PropTypes.func.isRequired,
  deleteRecipe: React.PropTypes.func.isRequired,
  deleteAllowed: React.PropTypes.bool.isRequired
}

export default MainPaneDumb
