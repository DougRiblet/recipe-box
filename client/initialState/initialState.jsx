const initialState = {
  active: '101010',
  mode: 'display',
  recipes: [
    {
      id: '101010',
      title: 'Grilled Cheese Sandwich',
      ingredients: '2 slices bread\n\n1 slice cheese\n\n2 dollops butter',
      directions: 'Spread butter on bread\n\nPlace cheese between bread slices\n\nGrill in pan until bread is lightly browned on both sides',
      notes: 'Yum yum!'
    }
  ]
}

export default initialState
