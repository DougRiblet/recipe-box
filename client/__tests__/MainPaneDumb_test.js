import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import MainPaneDumb from '../components/MainPaneDumb'
import initialState from '../initialState/initialState'

describe('MainPaneDumb', () => {
  test('Snapshot', () => {
    const MainPaneDumb_tree = renderer.create(
      <MainPaneDumb
        mode={initialState.mode}
        recipe={initialState.recipes[0]}
        deleteAllowed={true}
        updateRecipe={() => updateRecipe()}
        setMode={() => setMode()}
        deleteRecipe={() => deleteRecipe()}
      />
    ).toJSON()
    expect(MainPaneDumb_tree).toMatchSnapshot()
  })

  test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<MainPaneDumb
      mode={initialState.mode}
      recipe={initialState.recipes[0]}
      deleteAllowed={true}
      updateRecipe={() => updateRecipe()}
      setMode={() => setMode()}
      deleteRecipe={() => deleteRecipe()}
    />, div)
  })

  test('should display proper headings', () => {
    const sampleComponent = shallow(<MainPaneDumb
      mode={initialState.mode}
      recipe={initialState.recipes[0]}
      deleteAllowed={true}
      updateRecipe={() => updateRecipe()}
      setMode={() => setMode()}
      deleteRecipe={() => deleteRecipe()}
    />).html()
    const recipeHeadIng = '<h4>Ingredients</h4>'
    const recipeHeadDir = '<h4>Directions</h4>'
    const recipeHeadNot = '<h4>Notes</h4>'
    expect(sampleComponent).toContain(recipeHeadIng)
    expect(sampleComponent).toContain(recipeHeadDir)
    expect(sampleComponent).toContain(recipeHeadNot)
  })

  test('should display text of passed recipe', () => {
    const sampleComponent = shallow(<MainPaneDumb
      mode={initialState.mode}
      recipe={initialState.recipes[0]}
      deleteAllowed={true}
      updateRecipe={() => updateRecipe()}
      setMode={() => setMode()}
      deleteRecipe={() => deleteRecipe()}
    />).html()
    const recipeTtl = '<h1>Grilled Cheese Sandwich</h1>'
    const recipeIng = '<p>2 slices bread</p>'
    const recipeDir = '<p>Grill in pan until bread is lightly browned on both sides</p>'
    const recipeNot = '<p>Yum yum!</p>'
    expect(sampleComponent).toContain(recipeTtl)
    expect(sampleComponent).toContain(recipeIng)
    expect(sampleComponent).toContain(recipeDir)
    expect(sampleComponent).toContain(recipeNot)
  })

  test('should display edit button', () => {
    const sampleComponent = shallow(<MainPaneDumb
      mode={initialState.mode}
      recipe={initialState.recipes[0]}
      deleteAllowed={true}
      updateRecipe={() => updateRecipe()}
      setMode={() => setMode()}
      deleteRecipe={() => deleteRecipe()}
    />).html()
    const editButton = '<button class="setModeButton">Edit Recipe</button>'
    expect(sampleComponent).toContain(editButton)
  })
})
