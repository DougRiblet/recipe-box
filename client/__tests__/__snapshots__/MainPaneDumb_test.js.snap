// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPaneDumb Snapshot 1`] = `
<div
  id="contentDisplay"
>
  <h1>
    Grilled Cheese Sandwich
  </h1>
  <div
    id="ing-column"
  >
    <h4>
      Ingredients
    </h4>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>2 slices bread</p>
<p>1 slice cheese</p>
<p>2 dollops butter</p>
",
        }
      }
    />
  </div>
  <div
    id="dir-column"
  >
    <h4>
      Directions
    </h4>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Spread butter on bread</p>
<p>Place cheese between bread slices</p>
<p>Grill in pan until bread is lightly browned on both sides</p>
",
        }
      }
    />
    <h4>
      Notes
    </h4>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Yum yum!</p>
",
        }
      }
    />
  </div>
  <div
    className="mainPaneButtons"
  >
    <button
      className="setModeButton"
      onClick={[Function]}
    >
      Edit Recipe
    </button>
  </div>
</div>
`;
